airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -128,18 +128,28 @@ def _create_session_with_assume_role(self, session_kwargs: Dict[str, Any]) -> bo
             )
         session = botocore.session.get_session()
         session._credentials = credentials
-        region_name = self.basic_session.region_name
-        session.set_config_variable("region", region_name)
+        if self.basic_session is not None:
+            region_name = self.basic_session.region_name
+            session.set_config_variable("region", region_name)
         return boto3.session.Session(botocore_session=session, **session_kwargs)
 
     def _refresh_credentials(self) -> Dict[str, Any]:
         self.log.info('Refreshing credentials')
         assume_role_method = self.extra_config.get('assume_role_method', 'assume_role')
         sts_session = self.basic_session
         if assume_role_method == 'assume_role':
+            if sts_session is None:
+                raise Exception(
```

Comment:
RuntimeError or AirflowException is better; a bare Exception is too vague.

Reply:
yeah. RuntimeError. I think AirlfowException i kinda useless as it bears no more information than `Exception`.

Reply:
Yes, we overuse AirflowException way too much